pkg/loader/loader.go

```diff
@@ -60,19 +63,24 @@ type ModulesLoader interface {
 
 	// GetLoaded returns a mapping of loaded module paths to their text context.
 	GetLoaded() map[string]string
+
+	// GetLoadedModuleVersion returns the module version given the module name.
+	GetLoadedModuleVersion(moduleName string) string
 }
 
 // Module represents a starlark modules.
 type Module struct {
 	globals starlark.StringDict
 	data    []byte
+	version string
 	err     error
 }
 
 // ModulesLoader supports loading modules. In Starlark, each file is a module.
 type modulesLoader struct {
 	baseDir         string
 	loaded          map[string]*Module
+	loadedVersions  map[string]string
```

Comment:
Since `Module` already has the version as attribute, do we still need the `loadedVersions` map here? We would get the same thing via `.loaded["module"].version`

Reply:
The two keys are different.
`loadedVersions` is keyed on the simple module name (example if we load `@module//addon/xyz.ipd` this is just `module`)
`loaded ` is keyed on the entire name (`fullModule` so `@module//addon/xyz.ipd` in the example)
This will make it hard to connect with a file in main.ipd that is added to addons dict with loaded path like `@module//addon/addon.ipd` since it will not match with the key in loaded.

Reply:
refactored this to be simpler